src/commands/get/status.js

```diff
@@ -97,25 +96,47 @@ class StatusCommand extends Command {
     const { flags, args } = this.parse(StatusCommand);
     const { pkg, version } = this.parsePackage(args.package);
     const route = flags.events ? 'status-events' : 'status';
-    this.config = this.mergeConfig(flags);
-    const { statusHost: host } = this.config;
+    const { statusHost, wrhsHost, auth } = this.mergeConfig(flags);
 
-    if (!host) {
+    if (!statusHost) {
       this.error(this.missingHostError('status'));
     }
 
+    const wrhs = new Warehouse({
+      uri: `https://${auth.user}:${auth.pass}@${wrhsHost}`,
+      statusUri: `https://${auth.user}:${auth.pass}@${statusHost}`
+    });
+
+    const get = flags.events ? 'events' : 'get';
+
     try {
-      const response = await this.getWrhs(host, `/${route}/${encodeURIComponent(pkg)}/${args.env}/${version ? version : ''}`);
+      const response = new Promise((res, rej) => {
+        debug('Getting build status %s for %s@%s in %s', flags.events ? 'events' : 'information', pkg, version || 'HEAD', args.env);
+        wrhs.status[get]({ pkg, env: args.env, version }, function (err, data) {
+          if (err) return rej(err);
+
+          res(data);
+        });
+      });
 
       if (flags.json) {
-        return this.log(JSON.stringify(response));
+        return this.log(JSON.stringify(await response));
```

Comment:
don't await multiple times, you can await your new promise right away so there's less changes here.

Reply:
and thenify-ing it might make it seem less gross as well ;)